src/main/java/com/oltpbenchmark/api/Worker.java

```diff
@@ -182,170 +184,202 @@ synchronized public void cancelStatement() {
                     // Once a latency run is complete, we wait until the next
                     // phase or until DONE.
-                    continue work;
-                default:
-                    // Do nothing
+                    LOG.warn("preState is {}? will continue...", preState);
+                    continue;
+                }
+                default -> {
+                }
+                // Do nothing
             }
 
             // PART 3: Execute work
 
-            // TODO: Measuring latency when not rate limited is ... a little
-            // weird because if you add more simultaneous clients, you will
-            // increase latency (queue delay) but we do this anyway since it is
-            // useful sometimes
+            TransactionType transactionType = getTransactionType(pieceOfWork, prePhase, preState, workloadState);
 
-            long start = pieceOfWork.getStartTime();
+            if (!transactionType.equals(TransactionType.INVALID)) {
 
-            TransactionType type = invalidTT;
-            try {
-                type = doWork(pieceOfWork);
-            } catch (IndexOutOfBoundsException e) {
-                if (phase.isThroughputRun()) {
-                    LOG.error("Thread tried executing disabled phase!");
-                    throw e;
-                }
-                if (phase.getId() == this.state.getCurrentPhase().getId()) {
-                    switch (preState) {
-                        case WARMUP:
-                            // Don't quit yet: we haven't even begun!
-                            phase.resetSerial();
-                            break;
-                        case COLD_QUERY:
-                        case MEASURE:
-                            // The serial phase is over. Finish the run early.
-                            state.signalLatencyComplete();
-                            LOG.info("[Serial] Serial execution of all" + " transactions complete.");
-                            break;
-                        default:
-                            throw e;
+                // TODO: Measuring latency when not rate limited is ... a little
+                // weird because if you add more simultaneous clients, you will
+                // increase latency (queue delay) but we do this anyway since it is
+                // useful sometimes
+
+                // Wait before transaction if specified
+                long preExecutionWaitInMillis = getPreExecutionWaitInMillis(transactionType);
+
+                if (preExecutionWaitInMillis > 0) {
+                    try {
+                        LOG.debug("{} will sleep for {} ms before executing", transactionType.getName(), preExecutionWaitInMillis);
+
+                        Thread.sleep(preExecutionWaitInMillis);
+                    } catch (InterruptedException e) {
+                        LOG.error("Pre-execution sleep interrupted", e);
                     }
                 }
```

Comment:
I'm okay with adding this check here for every txn invocation in this PR, but please create a new issue so that we can add the boolean flag to enable/disable.

Reply:
will address here... #86